datafusion/core/src/physical_plan/file_format/parquet/page_filter.rs

```diff
@@ -110,14 +110,16 @@ impl PagePruningPredicate {
     pub fn try_new(expr: &Expr, schema: SchemaRef) -> Result<Self> {
         let predicates = split_conjunction(expr)
             .into_iter()
-            .filter_map(|predicate| match predicate.to_columns() {
-                Ok(columns) if columns.len() == 1 => {
-                    match PruningPredicate::try_new(predicate.clone(), schema.clone()) {
-                        Ok(p) if !p.allways_true() => Some(Ok(p)),
-                        _ => None,
+            .filter_map(|predicate| {
+                match PruningPredicate::try_new(predicate.clone(), schema.clone()) {
+                    Ok(p)
+                        if (!p.allways_true())
+                            && (p.required_columns().n_columns() < 2) =>
```

Comment:
This is a behavior change for `n_columns() == 0`. Based on:

```
    pub fn allways_true(&self) -> bool {
        self.predicate_expr
            .as_any()
            .downcast_ref::<Literal>()
            .map(|l| matches!(l.value(), ScalarValue::Boolean(Some(true))))
            .unwrap_or_default()
    }
```

I ran the test suite, panicing if `n_columns() == 0` and I can't get it to happen, so I guess it LGTM.
I assume that would default to false, in which case I think we'd want to return a `None` here?

Reply:
Yeah, we skip the predicate if we don't refer to any column. However you might be right (at least this is how I read your comment) that we need additional test coverage for a "constant" predicate (i.e. one that doesn't reference any column). I'll check next week if such a test exists and if not, add one.